helix-term/src/commands.rs

```diff
@@ -126,11 +128,145 @@ fn align_view(doc: &Document, view: &mut View, align: Align) {
     view.first_line = line.saturating_sub(relative);
 }
 
-/// A command is a function that takes the current state and a count, and does a side-effect on the
-/// state (usually by creating and applying a transaction).
-pub type Command = fn(cx: &mut Context);
+/// A command is composed of a static name, and a function that takes the current state plus a count,
+/// and does a side-effect on the state (usually by creating and applying a transaction).
+#[derive(Clone)]
+pub struct Command(fn(cx: &mut Context), &'static str);
 
-pub fn move_char_left(cx: &mut Context) {
+macro_rules! commands {
+    ( $($name: ident),* ) => {
```

Comment:
```suggestion
    ( $($name:ident),* ) => {
```
I think this is seen more in rust ecosystem, `: ` seemed weird.